src/LicenseKey.php

```diff
@@ -0,0 +1,233 @@
+    private const KEY_ORDER_ID = 'order_id';
+    private const KEY_STATUS = 'status';
+    private const KEY_EXPIRES_AT = 'expires_at';
+    private const KEY_CREATED_AT = 'created_at';
+    private const KEY_UPDATED_AT = 'updated_at';
+
+    /**
+     * The table associated with the model.
+     *
+     * @var string
+     */
+    protected $table = 'lemon_squeezy_license_keys';
+
+    protected $guarded = [];
+
+    /**
+     * The attributes that should be cast to native types.
+     *
+     * @var array
+     */
+    protected $casts = [
+        'activation_limit' => 'integer',
+        'instances_count' => 'integer',
+        'expired_at' => 'datetime',
+    ];
+
+    /**
+     * @throws MalformedDataError
+     */
+    public static function fromPayload(array $payload): LicenseKey {
+        $validator = Validator::make(array_merge([self::KEY_ID => $payload[self::KEY_ID]], $payload['attributes']), [
+            self::KEY_ID                => ['required', 'string'],
+            self::KEY_KEY               => ['required', 'string'],
+            self::KEY_KEY_SHORT         => ['required', 'string'],
+            self::KEY_ACTIVATION_LIMIT  => ['required', 'numeric'],
+            self::KEY_PRODUCT_ID        => ['required'],
+            self::KEY_ORDER_ID          => ['required'],
+            self::KEY_STATUS            => ['required'],
+            self::KEY_CREATED_AT        => ['required', 'date'],
+        ]);
+
+        if (!$validator->passes()) {
+            throw MalformedDataError::forLicenseKey($validator);
+        }
+
+        $attributes = $payload['attributes'];
+
+        $licenseKey = LicenseKey::create([
+            'lemon_squeezy_id' => $payload[self::KEY_ID],
+            'status' => $attributes[self::KEY_STATUS],
+            'disabled' => $attributes[self::KEY_DISABLED] ?? false,
+            'license_key' => $attributes[self::KEY_KEY],
+            'product_id' => $attributes[self::KEY_PRODUCT_ID],
+            'order_id' => $attributes[self::KEY_ORDER_ID],
+            'activation_limit' => $attributes[self::KEY_ACTIVATION_LIMIT],
+            'instances_count' => $attributes[self::KEY_INSTANCES_COUNT] ?? 0,
+            'expires_at' => isset($attributes[self::KEY_EXPIRES_AT])
+                ? Carbon::make($attributes[self::KEY_EXPIRES_AT])
+                : null,
+            'created_at' => $attributes[self::KEY_CREATED_AT],
+            'updated_at' => isset($attributes[self::KEY_UPDATED_AT])
+                ? Carbon::make($attributes[self::KEY_UPDATED_AT]) : null,
+        ]);
+
+        return $licenseKey;
+    }
+
+    /**
+     * The order this license key was generated for
+     */
+    public function order(): BelongsTo {
+        return $this->belongsTo(Order::class);
+    }
+
+    /**
+     * The billable that purchased the license
+     */
+    public function billable(): MorphTo {
+        /** @var Order $order */
+        $order = $this->order();
+
+        return $order->billable();
```

Comment:
Apparently this does not work. @driesvints how does one do this in Laravel?

Reply:
@driesvints no longer maintains, you can tag @JustSteveKing or me..Taking a look

Reply:
Hi @paulinevos, it should work. what error are you getting?

Reply:
I think it might be a type hint issue. This happens when the WebhookController dispatches the license key updated/created events and tries to pass along the billable:

```
Internal server error: Call to undefined method Illuminate\Database\Eloquent\Relations\BelongsTo::billable()
```

This is the webhook response we send back.

Reply:
Oh okay, try 
```php
 $order = $this->order->first();

 return $order->billable();
```

or 
```php
 return $this->order->billable();
```

Reply:
Just a quick update: it did actually work, it was more an issue with the order not existing in my dev database and it trying to chain onto `null` instead of throwing a not found exception. Will update asap

Reply:
Ok sorry it took a while, I was a bit swamped. 

So I solved this by just creating a foreign key for the order to ensure the order actually exists. Should be all good to go now :)